build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/CheckstylePrecommitPlugin.java

```diff
@@ -97,14 +95,17 @@ public void execute(Task task) {
             t.reports(r -> r.getHtml().getRequired().set(false));
         });
 
-        return checkstyleTask;
-    }
+        // Configure checkstyle tasks with an empty classpath to improve build avoidance.
+        // It's optional since our rules only rely on source files anyway.
+        project.getExtensions()
```

Comment:
I had to register this funny way to get around configuration ordering issues related to additional added source sets. Just using `tasks.withType().configureEach()` resulted in additional source sets (like javaRestTests and internalClusterTests) would have this config blasted away by the defaults in the convention plugin :shrug: